stats.cpp

```diff
@@ -1,5 +1,87 @@
+#include <bits/stdc++.h>
 #include "stats.h"
+int getNaNcount(const std::vector<double>& numericaldata)
+{   
+    int count=0;
+    for (auto& ptr : numericaldata) 
+        { 
+        if(std::isnan(ptr))
+            count++;
+        }
+    return count;
+}
 
-Stats Statistics::ComputeStatistics(const std::vector<___>& ) {
+std::vector<double> getNaNremoveddata(const std::vector<double>& numericaldata)
+{
+        std::vector<double> nanremoved;
+            for(double num : numericaldata)
+            {
+                if(!std::isnan(num))
+                    nanremoved.push_back(num);
+            }
+    return nanremoved;
+}
+double calculateAverage(const std::vector<double>& data)
+{
+    double sum = 0;
+    int noofdata = data.size();
+    for(double itr : data)
+        sum = sum + itr;
+    return sum/noofdata;
+}
+
+double findMax(const std::vector<double>& data)
+{
+    double max = INT_MIN;
```

Comment:
This code will fail if all the numbers are less than INT_MIN (can happen because the data is double, which can take values less than integer-minimum and greater than integer-max).
So this code will not work if all numbers are less than INT_MIN. Try making a test case for that to show the failure.
Then use the build-in stl max and min functions.